src/useBodySize.ts

```diff
@@ -0,0 +1,34 @@
+import { useMemo } from 'react';
+import useIsomorphicLayoutEffect from './useIsomorphicLayoutEffect';
+import useRafState from './useRafState';
+import { isClient } from './util';
+
+const useBodySize = (initialWidth = Infinity, initialHeight = Infinity) => {
+  const [state, setState] = useRafState<{ width: number; height: number }>({
+    width: isClient ? document.body.clientWidth : initialWidth,
+    height: isClient ? document.body.clientHeight : initialHeight,
+  });
+
+  const observer = useMemo(
+    () =>
+      new (window as any).ResizeObserver((entries) => {
```

Comment:
Please correct me if I'm wrong.
`useMemo()` will also be called on SSR requests, won't it? If yes then this line of code might throw on SSR request.